syntax = "proto3";

package kv;

option cc_generic_services = true; // 开启stub服务

service KVService {
  rpc Put (PutRequest) returns (PutResponse);
  rpc Get (GetRequest) returns (GetResponse);
}

message PutRequest {
  bytes key = 1;
  bytes value = 2;
  int32 version = 3;
}

message PutResponse {
  bool success = 1;
  bytes error = 2; // e.g., "ErrVersion"
}

message GetRequest {
  bytes key = 1;
}

message GetResponse {
  bytes value = 1;
  int32 version = 2;
  bytes error = 3; // e.g., "ErrNoKey"
}
